engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,61 +1531,184 @@ private static boolean serializeAllTable(
 
         final String[] columnSources = table.getDefinition().getColumnNamesArray();
 
-        try (final SharedContext sharedContext =
-                (columnSources.length > 1) ? SharedContext.makeSharedContext() : null) {
-            for (int ii = 0; ii < columnSources.length; ++ii) {
-                if (concurrentAttemptInconsistent()) {
-                    if (log.isDebugEnabled()) {
-                        final LogEntry logEntry = log.debug().append(System.identityHashCode(logIdentityObject))
-                                .append(" Bad snapshot before column ").append(ii);
-                        appendConcurrentAttemptClockInfo(logEntry);
-                        logEntry.endl();
-                    }
+        // Snapshot empty columns serially, and collect indices of non-empty columns
+        final int numColumnsToSnapshot =
+                (columnsToSnapshot != null) ? columnsToSnapshot.cardinality() : columnSources.length;
+        final TIntList nonEmptyColumnsIndices = new TIntArrayList(numColumnsToSnapshot);
+        if (!snapshotEmptyColumns(columnSources, columnsToSnapshot, table, logIdentityObject, snapshot,
+                nonEmptyColumnsIndices)) {
+            return false;
+        }
+        if (!nonEmptyColumnsIndices.isEmpty()) {
```

Comment:
While this is OK, _and_, while we tend to have less than 1M columns, although it's annoying to iterate over a bit set, I believe it's significantly nicer on cpu and mem usage.

Maybe you could flip bits in a copy instead of building the TIntList? Just some food for thought; I think I'd try to stick to BitSet since it's compact and fast to iterate.